src/core/or/connection_or.c

```diff
@@ -2434,8 +2461,7 @@ connection_or_send_netinfo,(or_connection_t *conn))
   cell_t cell;
   time_t now = time(NULL);
   const routerinfo_t *me;
-  int len;
-  uint8_t *out;
+  int r = 0;
```

Comment:
A trick that I like is set this to `-1` so then every `goto` will have the right error value set. Then if you reach the `cleanup:` label, just before you set `r = 0`. This way, any code addition later, we avoid the error of someone forgetting to set `r` within the error path.

Reply:
Did so in e9b6b8a15aa483de0c4308bef27602f2aeba2f66.